java/ddlogapi/DDlogAPI.java

```diff
@@ -190,27 +192,54 @@ public int commit() {
         return DDlogAPI.ddlog_transaction_commit(this.hprog);
     }
 
-    public int commit_dump_changes(Consumer<DDlogCommand> callback) {
+    public int commitDumpChanges(Consumer<DDlogCommand<DDlogRecord>> callback) {
         String onDelta = callback == null ? null : "onDelta";
         this.deltaCallback = callback;
         return this.ddlog_transaction_commit_dump_changes(this.hprog, onDelta);
     }
 
-    /// Callback invoked from commit_dump_changes.
+    // Callback invoked from commit_dump_changes.
     void onDelta(int tableid, long handle, boolean polarity) {
         if (this.deltaCallback != null) {
             DDlogCommand.Kind kind = polarity ? DDlogCommand.Kind.Insert : DDlogCommand.Kind.DeleteVal;
             DDlogRecord record = DDlogRecord.fromSharedHandle(handle);
-            DDlogCommand command = new DDlogCommand(kind, tableid, record);
+            DDlogRecCommand command = new DDlogRecCommand(kind, tableid, record);
             this.deltaCallback.accept(command);
         }
     }
 
+    // The FlatBufDescr class and commitDumpChangesToFlatbuf method are not
+    // meant to be invoked directly by user code; they are only used by the
+    // autogenerated FlatBuffer API code.
+    public static class FlatBufDescr {
+        public FlatBufDescr() {
+            this.buf = null;
+            this.size = 0;
+            this.offset = 0;
+        }
+        public void set(ByteBuffer buf, long size, long offset) {
+            this.buf = buf;
```

Comment:
Can you make some sanity checks, e.g., capacity is large enough?

Reply:
This class only exists because you cannot return a tuple in Java.  It does not implement any useful abstraction and is not meant to be used directly by the client (it does have to be public, unfortunately, because auto-generated Java code that uses this API lives in a different package). I don't think it needs to be complicated with sanity checks.